### PR TITLE
Migrate configuration page to Bootstrap

### PR DESCRIPTION
**Pull request by DavidKang**
We migrate the configuration page to Bootstrap.

screenshot_2019-02-21 configuration - open build service

### CODE REVIEW

**Comment by DavidKang** [comment]
@bgeuken, could you review it again? :smile_cat:

**Comment by DavidKang** [comment]
> Looks nice, but please add some `required` tags to the required fields

Thanks, I completely forgot about that :smile:

File: src/api/app/views/webui2/webui/configuration/index.html.haml [hunk 1]
```diff
@@ -3,4 +3,38 @@
...
+          .form-group
+            = f.label :title
+            = f.text_field :title, required: true, class: 'form-control'
+          .form-group
+            = f.label :description
+            ~ f.text_area :description, rows: 7, required: true, class: 'form-control'
+        .col-xl-6.mb-2
+          .h5 Base Configuration
+          %p.text-muted
+            %i.fas.fa-exclamation-circle.text-warning
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/configuration/index.html.haml]: It should be `fa-exclamation-triangle` according with `text-warning` according to our [patterns library](https://obs-patterns.netlify.com/atoms/icons.html)

**Comment by hennevogel** [comment]
The standard book on this is [Web Form Design](https://learning.oreilly.com/library/view/web-form-design/9781457102387/)  by Luke Wroblewski.

File: src/api/app/views/webui2/webui/configuration/index.html.haml [hunk 2]
```diff
@@ -3,4 +3,38 @@
...
+  .card-body#configuration
+    %h3.mb-4 Configure your Open Build Service instance
+    = form_for(@configuration, url: { action: :update }) do |f|
+      .row
+        .col-xl-6.mb-2
+          .h5 Frontpage Greeting
+          %p.text-muted
+            Change the text that is displayed as greeting on the frontpage
+          .form-group
+            = f.label :title
```
[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/views/webui2/webui/configuration/index.html.haml]: @DavidKang This should be addressed.

**Comment by DavidKang** [comment]
@hennevogel, Iwill have a look into the book. Thanks 😊

File: src/api/app/views/webui2/webui/configuration/index.html.haml (hunk 2 above)
[the remark this answers is not in the document]
Comment (DavidKang) [reply src/api/app/views/webui2/webui/configuration/index.html.haml]: @dmarcoux, addressed, thanks :smile_cat:

File: src/api/app/assets/stylesheets/webui2/configuration.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+@include media-breakpoint-up(xl) {
```
Comment (dmarcoux) [inline src/api/app/assets/stylesheets/webui2/configuration.scss]: This file isn't needed anymore, right?

Comment (DavidKang) [reply src/api/app/assets/stylesheets/webui2/configuration.scss]: right!

File: src/api/app/views/webui2/webui/configuration/index.html.haml [hunk 4]
```diff
@@ -4,3 +4,45 @@
...
+            = f.label :description
+            %abbr.text-danger{ title: 'required' } *
+            ~ f.text_area :description, rows: 7, required: true, class: 'form-control'
+
+      .h5.mt-4 Base Configuration
+      %p.text-muted
+        %i.fas.fa-exclamation-triangle.text-warning
+        Backend restart is needed after a change here.
+      .row
+        .col-lg-4
```
Comment (hennevogel) [inline src/api/app/views/webui2/webui/configuration/index.html.haml]: This still is a bad idea. Do not scale forms horizontal.

**Pull request merged**